src/lib/components/submit/Submit.svelte

```diff
@@ -46,8 +46,7 @@
 			return;
 		}
 
-		// TODO: select step should only appear if governance canister is NNS.
-		step = 'select';
+		step = $governanceTypeStore === 'icp' ? 'select' : 'write';
```

Comment:
If you want to assign it in the init function, this is not guaranteed to work 100% of the time. It's possible that the routeKey is known but the governanceTypeStore is not yet derived, therefore we should implement another solution.

You can add a TODO and solve it in another PR, that way we can move forward.